app/controllers/cancerdata.controller.js

```diff
@@ -0,0 +1,189 @@
+const { dbCredentials } = require("../config/oncodb.config.js");
+const { dataBaseQueryHelper } = require("databasequeryhelper.js");
+const { removeNewlinesAndUnderscores, changeSpecialCharsToBlank, cleanUpTranslator, convertToUnderscores } = require("../utilities/parsingFunctions.js");
+
+const events = require('events');
+const readline = require('readline');
+var qs = require('querystring');
+const fs = require('fs');
+
+async function testQuery(req, res){
+  try {
+    const resultOfQuery = await dbCredentials.query("SELECT * FROM GBM_TCGA_SPLICE LIMIT 3;");
+    res.send(resultOfQuery);
+    /*res.json({
+    	data: resultOfQuery
+    })*/
+  } catch (error) {
+    res.send([4, 5, 6]);
+  }
+}
+
+async function getUiFields(req, res){
+	if (req.method == 'POST') {
+	    var body = '';
+
+	    req.on('data', function (data) {
+	        body += data;
+
+	        if (body.length > 1e7)
+	                request.connection.destroy();
+	    });
+
+	    const outputObject = {};
+	    var currentFileContents = "";
+	    var content = "";
+	    var refname = "";
+
+		req.on('end', function () {
+	            var post = qs.parse(body);
+
+	            const queryHelperMap = dataBaseQueryHelper(post["cancerName"]);
+	            const clinicalMetadataResult = await dbCredentials.query(queryHelperMap["META"]["QUERY"]);
+	            clinicalMetadataResult.fields.forEach((element) => {
+	            	const fieldname = removeNewlinesAndUnderscores(element.name);
+	            	outputObject["meta"][fieldname] = "";
+	            });
+	            
+				fs.readdir(queryHelperMap["META"]["COLUMNS"], (err, files) => {
```

Comment:
Remove text files and make smaller functions or mappings instead to get the ranges/gender etc